packages/react-form-state/src/FormState.ts

```diff
@@ -360,31 +383,39 @@ export default class FormState<
   }
 
   private updateRemoteErrors(errors: RemoteError[]) {
-    this.setState(({fields}: State<Fields>) => {
-      const errorDictionary = errors.reduce(
-        (accumulator: any, {field, message}) => {
-          if (field == null) {
-            return accumulator;
-          }
-
-          return set(accumulator, field, message);
-        },
-        {},
-      );
-
+    this.setState(({fields}: State<Fields>, {externalErrors = []}) => {
+      const fieldWithErrorMapper = fieldWithErrorMapperFactory([
+        ...errors,
+        ...externalErrors,
+      ]);
       return {
         errors,
-        fields: mapObject(fields, (field, path) => {
-          return {
-            ...field,
-            error: errorDictionary[path],
-          };
-        }),
+        fields: mapObject(fields, fieldWithErrorMapper),
       };
     });
   }
 }
 
+function errorDictionaryFactory(errors: RemoteError[]) {
```

Comment:
I don't really like these names, we should name functions for what they *do* not what they *are*.

Reply:
IMO, factory pattern is well known pattern.
This function transforms array into object format.

Reply:
I think something like `getErrorDictionary` is common.

Reply:
Factory is a well known pattern but naming functions with a factory suffix is not. Half of the functions in a given JS codebase are technically factories (they output custom objects based on input), but we don't name everything `translationComponentFactory`, `apolloConnectedComponentFactory`, `fixtureFactory`. We tend to name them based on what they do instead.

Reply:
For example: if it transforms arrays into an object format we should call it something like `mapErrorsToDictionary`.